core/src/cpu/air/mod.rs

```diff
@@ -420,6 +434,53 @@ impl CpuChip {
             .assert_eq(local.pc, next.pc);
     }
 
+    /// Constraints related to the commit instruction.
+    pub(crate) fn commit_eval<AB: SP1AirBuilder>(
+        &self,
+        builder: &mut AB,
+        local: &CpuCols<AB::Var>,
+        is_commit: AB::Expr,
+    ) {
+        let public_values = builder.public_values();
+
+        // Convert public values into words.
+        let mut digest_words = Vec::new();
+        for bytes in public_values.chunks_exact(WORD_SIZE) {
+            let bytes_expr_vec: Vec<AB::Expr> =
+                bytes.iter().map(|byte| (*byte).into()).collect::<Vec<_>>();
+
+            digest_words.push(Word::<AB::Expr>(bytes_expr_vec.try_into().unwrap()));
+        }
+
+        // Get the ecall specific columns.
+        let ecall_columns = local.opcode_specific_columns.ecall();
+
+        // Verify the index bitmap.
+        let mut bitmap_sum = AB::Expr::zero();
+        for bit in ecall_columns.index_bitmap.iter() {
+            builder.when(local.selectors.is_ecall).assert_bool(*bit);
+            bitmap_sum += (*bit).into();
+        }
+        builder.when(is_commit.clone()).assert_one(bitmap_sum);
+
+        // Retrieve the public_values_digest_word to check against the one passed into the commit ecall.
```

Comment:
This also seems a bit jank TBH because you're using public values as a way to detect if it's an interaction builder. I wonder if there's a better way we can do it?

Maybe we should expose something like `builder.is_interaction_builder()`?